internal/infra/repository_memory.go

```diff
@@ -0,0 +1,45 @@
+package infra
+
+import (
+	"errors"
+	"github.com/Kenny201/go-yandex-shortener.git/internal/domain/shortener/aggregate"
+)
+
+type RepositoryMemory struct {
+	urls map[string]*aggregate.URL
+}
+
+func NewRepositoryMemory() *RepositoryMemory {
+	return &RepositoryMemory{
+		urls: make(map[string]*aggregate.URL),
+	}
+}
+
+func (rm *RepositoryMemory) Get(id string) (*aggregate.URL, error) {
+	if _, ok := rm.urls[id]; !ok {
+		err := errors.New("url not found")
+		return nil, err
+	}
+
+	return rm.urls[id], nil
+}
```

Comment:
Эту функцию можно немного оптимизировать, хотя доступ к элементу мапы по сложности O(1), все-таки хорошим тоном будет получать значение один раз:
```suggestion
	url, ok := rm.urls[id]
	if !ok {
		return nil, errors.New("url not found")
	}
	return url, nil
```